src/middleware/axum.rs

```diff
@@ -15,9 +20,20 @@ use tower::{Layer, Service};
 use tracing::Span;
 use tracing_opentelemetry_instrumentation_sdk::http as otel_http;
 
+/// Function type for filtering HTTP requests by path.
+///
+/// Takes a path string and returns true if the request should be traced.
 pub type Filter = fn(&str) -> bool;
+
+/// Function type for extracting string representation from a matched path type.
+///
+/// Used to convert Axum's matched path type to a string for span attributes.
 pub type AsStr<T> = fn(&T) -> &str;
 
+/// OpenTelemetry layer for Axum applications.
+///
+/// This layer provides automatic tracing instrumentation for Axum web applications,
+/// creating spans for HTTP requests with appropriate semantic attributes.
```

Comment:
We should include an example of how this `OtelAxumLayer` could be constructed and used.

Something like this should suffice:

```rust
let app = axum::Router::new()
    .nest("/api", api_routes)
    .layer(OtelAxumLayer::new(axum::extract::MatchedPath::as_str));
```